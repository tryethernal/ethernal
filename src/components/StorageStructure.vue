<template>
    <div>
        <Add-Tracked-Key-Modal ref="addTrackedKeyModal" />
        {{storage.label}}
        <v-btn @click.stop="openAddTrackedKeyModal(storage.index)" color="primary" icon variant="outlined" size="x-small" v-if="storage.children != null">
            <v-icon>mdi-plus</v-icon>
        </v-btn>
        <Storage-Structure v-bind:storage="child" v-for="(child, idx) in storage.children" :key="idx" class="ml-4" @addStorageStructureChild="addStorageStructureChild" />
    </div>
</template>
<script>
import AddTrackedKeyModal from './AddTrackedKeyModal';

export default {
    name: 'StorageStructure',
    props: ['storage'],
    components: {
        AddTrackedKeyModal
    },
    methods: {
        openAddTrackedKeyModal: function(structIndex) {
            this.$refs.addTrackedKeyModal
                .open({ variableIndex: structIndex })
                .then(res => {
                    if (res) {
                        this.addStorageStructureChild(this.storage, res.variableIndex, res.key);
                    }
                });
        },
        addStorageStructureChild: function(storage, idx, key) {
            this.$emit('addStorageStructureChild', storage, idx, key);
        }
    }
};
</script>
