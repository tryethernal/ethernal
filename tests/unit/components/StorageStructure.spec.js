import MockHelper from '../MockHelper';
import flushPromises from 'flush-promises';

import StorageStructure from '@/components/StorageStructure.vue';
import StorageProp from '../fixtures/StorageProp.json';

describe('StorageStructure.vue', () => {
    let helper;

    beforeEach(() => {
        helper = new MockHelper();
    });

    it('Should display the correct info', async () => {
        const wrapper = helper.mountFn(StorageStructure, {
            propsData: {
                storage: StorageProp
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
