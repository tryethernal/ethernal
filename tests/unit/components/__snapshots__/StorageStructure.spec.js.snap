// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StorageStructure.vue Should display the correct info 1`] = `
<div>
  <div class="v-dialog__container">
    <!---->
  </div>
  mapping(address =&gt; uint256) _deposits:
  <button type="button" class="v-btn v-btn--icon v-btn--outlined v-btn--round theme--light v-size--x-small primary--text"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-plus theme--light"></i></span></button>
  <div class="ml-4">
    <div class="v-dialog__container">
      <!---->
    </div>
    0x56713b5deeb3438c5fcd203a134d01609bD5463C;
    <!---->
  </div>
</div>
`;
